services/director-v2/src/simcore_service_director_v2/api/routes/dynamic_services.py

```diff
@@ -190,18 +190,26 @@ async def stop_dynamic_service(
 )
 @log_decorator(logger=logger)
 async def service_retrieve_data_on_ports(
+    request: Request,
+    node_uuid: NodeID,
     retrieve_settings: RetrieveDataIn,
-    service_base_url: URL = Depends(get_service_base_url),
-    services_client: ServicesClient = Depends(get_services_client),
-):
-    # the handling of client/server errors is already encapsulated in the call to request
-    resp = await services_client.request(
-        "POST",
-        f"{service_base_url}/retrieve",
-        data=retrieve_settings.json(by_alias=True),
-        timeout=httpx.Timeout(
-            5.0, read=60 * 60.0
-        ),  # this call waits for the service to download data
-    )
-    # validate and return
-    return RetrieveDataOutEnveloped.parse_obj(resp.json())
+    scheduler: DynamicSidecarsScheduler = Depends(get_scheduler),
+) -> RetrieveDataOutEnveloped:
+    try:
+        return await scheduler.retrive(node_uuid, retrieve_settings.port_keys)
```

Comment:
```suggestion
        return await scheduler.retrieve(node_uuid, retrieve_settings.port_keys)
```

Reply:
and a secondary question: why is the scheduler involved in retrieving the ports on the running services?
the scheduler takes the role of scheduling the services (start/stop/restart/pause/kill). why should it take care of calling the services'API?
it's minor, but I think I would recommend not to bundle too many functions in the scheduler.

Reply:
I've replied below